ui/src/components/EditAffiliationsModal/index.tsx

```diff
@@ -0,0 +1,218 @@
+                    >
+                        <div className="relative mx-8 my-20 inline-block w-11/12 transform overflow-hidden rounded-lg bg-white-50 px-4 pt-5 pb-4 text-left align-bottom shadow-xl transition-all sm:px-8 sm:py-6 sm:align-middle xl:max-w-5xl">
+                            <HeadlessUI.Dialog.Title
+                                as="h3"
+                                className="pb-8 text-center font-montserrat text-lg font-medium leading-6 text-grey-900"
+                            >
+                                Choose your affiliation(s) for this publication
+                            </HeadlessUI.Dialog.Title>
+
+                            <p className="mb-2 block text-sm leading-snug text-grey-700 transition-colors duration-500 dark:text-white-100">
+                                The following affiliations are present on{' '}
+                                <Components.Link
+                                    href={`https://orcid.org/my-orcid?orcid=${props.author?.user?.orcid}`}
+                                    openNew
+                                    className="text-teal-600 underline transition-colors duration-500 dark:text-teal-400"
+                                >
+                                    your ORCID® profile
+                                </Components.Link>
+                                . To add a new affiliation, please enter it on{' '}
+                                <Components.Link
+                                    href={`https://orcid.org/my-orcid?orcid=${props.author?.user?.orcid}`}
+                                    openNew
+                                    className="text-teal-600 underline transition-colors duration-500 dark:text-teal-400"
+                                >
+                                    your ORCID profile
+                                </Components.Link>
+                                .
+                            </p>
+
+                            <p className="mb-2 block text-sm leading-snug text-grey-700 transition-colors duration-500 dark:text-white-100">
+                                <Components.Link
+                                    href="https://support.orcid.org/hc/en-us/sections/360002054993-Assert-your-affiliations"
+                                    openNew
+                                    className="text-teal-600 underline transition-colors duration-500 dark:text-teal-400"
+                                >
+                                    Click here
+                                </Components.Link>{' '}
+                                to learn more about adding affiliations to ORCID.
+                            </p>
+
+                            <div className="pt-8">
+                                <Components.AuthorAffiliations
+                                    scrollHeight={400}
+                                    loading={!orcidAffiliations.length && isValidating}
```

Comment:
It looks like there is an isLoading property in SWR that could be used here instead of isValidating + checking the affiliations array: https://swr.vercel.app/docs/advanced/understanding#combining-with-isloading-and-isvalidating-for-better-ux

Reply:
<img width="859" alt="image" src="https://github.com/JiscSD/octopus/assets/48569671/cbafebe9-e42d-4bf9-b799-151710ce8c43">

Maybe they added 'isLoading' in a newer version of SWR. Current version we use doesn't have that prop

Reply:
_Comparing to the existing **isValidating** value, **isLoading is a new property** that can help you for the more general loading cases for UX._